graphqlapi/src/main/java/io/stargate/graphql/web/resources/GraphqlCache.java

```diff
@@ -223,6 +224,23 @@ public void onKeyspaceChanged(
     // It is assumed that the data model will only evolve by deploying new GraphQL schema versions.
   }
 
+  public void put(
+      String keyspaceName, SchemaSource newSource, GraphQL graphql, AuthenticationSubject subject) {
+    Map<String, String> headers = subject.customProperties();
+    DataStore dataStore = buildUserDatastore(subject, headers);
+    String decoratedKeyspaceName = persistence.decorateKeyspaceName(keyspaceName, headers);
+
+    Keyspace keyspace = dataStore.schema().keyspace(keyspaceName);
+    if (keyspace == null) {
+      LOG.trace("Keyspace {} does not exist", decoratedKeyspaceName);
+      return;
+    }
+    LOG.trace(
+        "Putting new schema version: {} for {}", newSource.getVersion(), decoratedKeyspaceName);
+    DmlGraphqlHolder schemaHolder = new DmlGraphqlHolder(newSource, keyspace, graphql);
+    dmlGraphqls.put(decoratedKeyspaceName, schemaHolder);
```

Comment:
@olim7t do you think that we can safely put the value, without this:
```
(currentHolder == null)
            ? dmlGraphqls.putIfAbsent(decoratedKeyspaceName, newHolder) == null
            : dmlGraphqls.replace(decoratedKeyspaceName, currentHolder, newHolder);
```
I think that we can `put()` it without checking the previous version because we are sure that if someone is deploying a new schema, it is the newest one. wdyt?

Reply:
Yes, think we are safe here. The LWT in `SchemaSourceDao` guarantees that a deployment will never be overwritten. The worst that can happen here is:
* thread A starts a deployment and writes the new row in `schema_source`
* thread B executes a GraphQL query for this keyspace; `getDml()` sees the new row and refreshes the entry
* thread A finishes the deployment, `put()` doesn't see that the new version is already computed, and computes it again

We can tolerate a double init in this corner case, it's just a few more CPU cycles and has no impact for users.